group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -983,27 +1085,81 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
         return new CoordinatorResult<>(records, response);
     }
 
+    private void replaceStaticMemberInConsumerGroup(
```

Comment:
nit: removeMemberAndCancelTimers? The logic is not tight to static members. I would also directly pass the groupId and the memberId as this is all it needs.

Reply:
done.